src/api/l_data.c

```diff
@@ -70,10 +76,9 @@ static int l_lovrDataNewSoundData(lua_State* L) {
     uint64_t frames = luaL_checkinteger(L, 1);
     uint32_t channels = luaL_optinteger(L, 2, 2);
     uint32_t sampleRate = luaL_optinteger(L, 3, 44100);
-    uint32_t format = luaL_optinteger(L, 4, 16);
+    SampleFormat format = luax_checkenum(L, 4, SampleFormat, "i16");
     Blob* blob = luax_totype(L, 5, Blob);
```

Comment:
This is probably just my tendency to condense everything too much, but if the Blob argument here was thought of as "contents", then then contents for your SoundData could be:

- `Blob`: You have data you want to use as the contents
- `nil`: You want empty contents initialized for you
- `'stream'` or `'append'`: You want to stream the contents to the SoundData using `:append`
- ...potentially others?

That would avoid the need for an extra newSoundDataStream function and just puts the word 'stream' at the end of this one :)

Reply:
well... I wouldn't phrase it as a "need" for an extra method, but an opportunity for a clearer API :P but that's just my Objective-C influenced opinion; it's your API, I can implement the above suggestion :)